File: content/snippets/scala2/interview_lcci/17/11_find_closest.scala
object Lcci1711FindClosest {
  def findClosest(words: Array[String], word1: String, word2: String): Int = {
    var i1 = -1
    var i2 = -1
    var ans = Int.MaxValue
    for (i <- words.indices) {
      if (words(i) == word1) i1 = i
      if (words(i) == word2) i2 = i
      if (i1 != -1 && i2 != -1) ans = math.min(ans, math.abs(i1 - i2))
    }
    ans
  }

  def main(args: Array[String]): Unit = {
    println(findClosest(Array("I", "am", "a", "student", "from", "a", "university", "in", "a", "city"), "a", "student")) // 1
    println(findClosest(Array("aa", "bb"), "aa", "bb")) // 1
  }
}

File: content/snippets/python/interview_lcci/17/11_find_closest.py
from typing import List


class Solution:
    def findClosest(self, words: List[str], word1: str, word2: str) -> int:
        i1 = i2 = -1
        ans = float('inf')
        for i, w in enumerate(words):
            if w == word1:
                i1 = i
            if w == word2:
                i2 = i
            if i1 != -1 and i2 != -1:
                ans = min(ans, abs(i1 - i2))
        return int(ans)


def main():
    sol = Solution()
    print(sol.findClosest(["I","am","a","student","from","a","university","in","a","city"], "a", "student"))  # 1
    print(sol.findClosest(["aa", "bb"], "aa", "bb"))  # 1


if __name__ == "__main__":
    main()
